doc/query.md

```diff
@@ -39,27 +39,60 @@ Secondly, these documents are in the datastore:
 ```java
 { "name": "mike", 
   "age": 12, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Mike goes to middle school and likes reading books." };
 
 { "name": "mike", 
   "age": 34, 
-  "pet": {"species": "dog"} };
+  "pet": {"species": "dog"},
+  "comment": "Mike is a doctor and likes reading books." };
 
 { "name": "fred", 
   "age": 23, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Fred works for a startup out of his home office." };
 ```
 
-### Creating indexes
+### Creating Indexes
 
 In order to query documents, indexes need to be created over
-the fields to be queried against.
+the fields to be queried against.  Currently we support two types of indexes.  The first, a JSON index, is used by query clauses containing comparison operators like `$eq`, `$lt`, `$gt`, etc.  Query clauses containing these operators rely on standard SQL syntax along with JSON indexes to provide query results.  The second, a TEXT index, uses SQLite's full text search (FTS) engine.  A query clause containing a `$text` operator with a `$search` operator uses [SQLite FTS SQL syntax] [ftsHome] along with a TEXT index to provide query results.
```

Comment:
Is `rely on standard SQL syntax along with JSON indexes` really required? Perhaps just say they are based on standard SQLite indexes?